Separate unused and unusable views

cymed opened the pull request:
Separating the old 2015 version views according to https://github.com/teksi/wastewater/issues/11#issuecomment-1822368211 from those views that work but are not automatically loaded. Also adds two views for the new wastewater structure types through create_views.py (see https://github.com/teksi/wastewater/discussions/38)

sjib commented:
To seperate unusable views I think is good.
Not so clear for me is the definition of unused. 
@cymed When is a view unused? What does this imply?

I think we should create for all subclasses views that integrate the subclass as it helps to add data more easily. For all forms we create that include sub/superclasses I think we also need superclass views.

cymed commented:
What I did was basically to separate the views moved in https://github.com/teksi/wastewater/pull/12
For me, unused views are those views that are not necessary for the functionality of the default qgis project, but can be useful for specific purposes. I.e. tww_app.vw_tww_maintenance_on_reach can be useful to visualize which reaches are affected by a maintenance event, but TWW works well without it.
I do agree though that some of the views in the unused_views folder are duplicates of singleInheritances in create_views.py

3nids commented:
just to clarify, files in unused folder means they are not loaded in the database, so it's unused/untested code at the moment.

cymed commented:
I quickly looked through them:, we have 4 folders now
- heritage_dss2015 for views referencing a table that no longer exists
- singleInheritance for views that could be made dynamically in create_views.py
- needs_fixing for views referencing fields that no longer exist
- working for views that can be directly implemented

vw_organisation is outside of all this because it was in qgep's create_views.py but was dropped now (MultiInheritance for the organisation_type) I am in favour of re-incorporating it because it allows an easier mapping of network elements to the type of owner/operator, which is necessary for certain visualisations.

sjib commented:
> vw_organisation is outside of all this because it was in qgep's create_views.py but was dropped now (MultiInheritance for the organisation_type) I am in favour of re-incorporating it because it allows an easier mapping of network elements to the type of owner/operator, which is necessary for certain visualisations.
> @urskaufmann @sjib what is your opinion on how to proceed?

The class organisation has in VSA-DSS 2020 no subclasses anymore and is based on the definition in https://www.vsa.ch/models/2020_1/SIA405_Base_Abwasser_1_2_d_LV95-20231018.ili, so it might not need an extra view anymore 

`  TOPIC Administration = 
   
   CLASS Organisation EXTENDS SIA405_BaseClass = 
      ATTRIBUTE
        !!@ comment = "Allgemeine Bemerkungen"
        Bemerkung: TEXT*255;
        !!@ comment = "Die Bezeichnung für Gemeinden wird gemäss BFS-Liste übernommen; für Private gemäss UID-Register, allenfalls mit Zusatz des Ortes der Filiale, falls keine separate UID dafür besteht."
        Bezeichnung: MANDATORY SIA405_Base_Abwasser_1_LV95.OrganisationBezeichnung;
        !!@ comment = "Offizielle Gemeinde Nummer gemäss Bundesamt für Statistik, bei Gemeinde zwingend, sonst leer lassen."
        Gemeindenummer: GemeindeNr;
        !!@ comment = "Kurzbezeichnung"
        Kurzbezeichnung: TEXT*12;
        Organisationstyp: MANDATORY (
          !!@ comment = "Abwasserverband"
          Abwasserverband,
          !!@ comment = "Teil einer Bundesverwaltung (z.B. Bundesamt für Umwelt, ASTRA, Armasuisse)"
          Bund,
          !!@ comment = "Gemeinde gemäss Bundesamt für Statistik"
          Gemeinde,
          !!@ comment = "Teil einer Gemeindeverwaltung, die eine Rolle übernimmt ((z.B. Geoinformation Stadt Bern) als Traegerschaft, Datenlieferant, etc. (aber nicht als Datenherr / (oeffentlicher) Eigentuemer)"
          Gemeindeabteilung,
          !!@ comment = "Genossenschaft oder Korporation: Körperschaft öffentlichen Rechts. Falls privaten Rechtes dann als Privat abbilden."
          Genossenschaft_Korporation,
          !!@ comment = "Teil einer kantonalen Verwaltung (z.B. Amt für Umweltschutz, Amt für Abwasserentsorgung)"
          Kanton,
          !!@ comment = "Privatperson oder Privatorganisation, welche im Rahmen der Entwässerungsplanung auftritt"
          Privat
        );
        !!@ comment = "Status der Organisation, damit untergegangende Organisationen nicht einfach gelöscht werden müssen und kontrolliert werden kann, ob noch Beziehungen auf untergegangene Organisationen, z.B. bei Gemeindefusion oder Konkurs einer Firma vorhanden sind"
        Status: MANDATORY (
          aktiv,
          untergegangen
        );
        !!@ comment = "Referenz zur Unternehmensidentifikation des Bundesamts fuer Statistik (www.uid.admin.ch), z.B. CHE123456789"
        UID: UID;
    UNIQUE 
        !!@comment = "!! Neben UNIQUE OID zusätzlich auch Kombination Bezeichnung, Organisationstyp, UID (Wegleitung GEP-Daten 2020)"
        Bezeichnung, Organisationstyp, UID; 
    END Organisation;`

How to handle organisations and how to import / export organisations should be defined with the implementation of the INTERLIS import/export interface (-> if needed we open an extra discussion for that).

urskaufmann commented:
table organisation: it works without view (because there are no subclasses), is already configured in the new project.

urskaufmann commented:
vw_drainage_system remains in dss_2020 (perhaps I have writen it on the wrong list, sorry). Should be moved from heritage_dss2015 to singleinheritance

urskaufmann commented:
in the database version 23.11.2023 I miss vw_catchment_area_wwp_connections

urskaufmann commented:
vw_catchment_area_additional: is a cool thing that helps a lot (less queries for users). 
I suggest to add in this vw also the calculated fields I miss: 

- "Fred" for all four discharge_coefficient (surface_area * discharge_coefficient/100 ->_Ared_wwc, _Ared_wwp, _Ared_rwc, _Ared_rwp) 
- population for two population_densities (surface_area * population_density -> _PE_current, _PE_planned). PE is used as shortcut in mike+ for "Einwohner".

urskaufmann commented:
vw_tww_maintenace_on_reach / _one_wws in "needs-fixing": to be correct, it should be vw_tww_maintenace_event_on_reach (very long, shortcut?).
For me very helpful layers, to be integrated in the project. 
I suggest to add some more fields:
- *_on_wws: also ws_renovation_necessity, ws_urgency_figure, ws_obj_id
- *_on_reach: also ws_renovation_necessity, ws_urgency_figure, re_clear_height, re_obj_id, ws_obj_id
in _on reach I integrated also ->reachpoints->wastewaternode_identifier to have "Knoten_oben", "Knoten_unten": to discuss

cymed commented:
> table organisation: it works without view (because there are no subclasses), is already configured in the new project.

moved to heritage

> vw_drainage_system remains in dss_2020 (perhaps I have writen it on the wrong list, sorry). Should be moved from heritage_dss2015 to singleinheritance in the next PR.

added to singleinheritance

> in the database version 23.11.2023 I miss vw_catchment_area_wwp_connections

see [here](https://github.com/teksi/wastewater/tree/main/datamodel/app/view/catchment_area)


> vw_catchment_area_additional: is a cool thing that helps a lot (less queries for users). I suggest to add in this vw also the calculated fields I miss:
> 
>     * "Fred" for all four discharge_coefficient (surface_area * discharge_coefficient/100 ->_Ared_wwc, _Ared_wwp, _Ared_rwc, _Ared_rwp)
> 
>     * population for two population_densities (surface_area * population_density -> _PE_current, _PE_planned). PE is used as shortcut in mike+ for "Einwohner".

Nice addition suggestion, can you provide the sql?
As your suggestion links to [the vw in the catchment area folder](https://github.com/teksi/wastewater/blob/main/datamodel/app/view/catchment_area/vw_catchment_area_additional.sql) and not [the vw in unused views](https://github.com/teksi/wastewater/blob/main/datamodel/app/view/unused_views/working/vw_tww_catchment_area.sql) the question remains if we can get rid of the latter.


> vw_tww_maintenace_on_reach / _one_wws in "needs-fixing": to be correct, it should be vw_tww_maintenace_event_on_reach (very long, shortcut?). For me very helpful layers, to be integrated in the project. I suggest to add some more fields:
> 
>     * *_on_wws: also ws_renovation_necessity, ws_urgency_figure, ws_obj_id
> 
>     * *_on_reach: also ws_renovation_necessity, ws_urgency_figure, re_clear_height, re_obj_id, ws_obj_id
>       in _on reach I integrated also ->reachpoints->wastewaternode_identifier to have "Knoten_oben", "Knoten_unten": to discuss

I was planning on implementing such a view. Fix is in the next PR.
